packages/pnpm-sync/src/start.ts

```diff
@@ -58,7 +65,23 @@ program
           if (lockfile === null) {
             return undefined;
           } else {
-            return lockfile;
+            const lockfilePackages: Record<string, ILockfilePackage> = {};
+            for (const packagePath in lockfile.packages) {
+              if (lockfile.packages[packagePath]) {
```

Comment:
The `packagePath` name is not very correct.  A better name might be `versionPath` (i.e. `.pnpm` folder subpath corresponding to the `"version"` field in from your `IVersionSpecifier`) or else `packagesKey` (i.e. a key from the `packages` object).

This kind of logic is often much more understandable if you include examples of the strings, something like this:

```ts
// Example versionPath: "/webpack-filter-warnings-plugin@1.2.1(webpack@4.47.0)"
for (const versionPath of Object.keys(lockfile.packages)) {}
```